### PR TITLE
WIP: Add scenario for plaintext with response caching

### PR DESCRIPTION
**Pull request by mikeharder**
@JunTaoLuo, @BrennanConroy

### CODE REVIEW

File: src/Benchmarks/Middleware/CachedPlaintextMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        private readonly RequestDelegate _next;
+
+        public CachedPlaintextMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public Task Invoke(HttpContext httpContext)
+        {
+            if (httpContext.Request.Path.StartsWithSegments(_path, StringComparison.Ordinal))
```
Comment (JunTaoLuo) [inline src/Benchmarks/Middleware/CachedPlaintextMiddleware.cs]: Is there no way to avoid this? Not sure how much overhead this adds.

Comment (mikeharder) [inline src/Benchmarks/Middleware/CachedPlaintextMiddleware.cs]: It should be fine, since it's done by all middlewares in Benchmarks including Plaintext.

Comment (JunTaoLuo) [inline src/Benchmarks/Middleware/CachedPlaintextMiddleware.cs]: Hmm I see that the other middlewares also uses the same pattern.

File: src/Benchmarks/Middleware/CachedPlaintextMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+        public Task Invoke(HttpContext httpContext)
+        {
+            if (httpContext.Request.Path.StartsWithSegments(_nocachePath, StringComparison.Ordinal))
+            {
+                return PlaintextMiddleware.WriteResponse(httpContext.Response);
+            }
+            else if (httpContext.Request.Path.StartsWithSegments(_path, StringComparison.Ordinal))
+            {
+                httpContext.Response.Headers["cache-control"] = "public, max-age=1";
```
Comment (JunTaoLuo) [inline src/Benchmarks/Middleware/CachedPlaintextMiddleware.cs]: Not really realistic, but we want to simulate entries that expire.

**Pull request merged**